test/EntityFramework/FunctionalTests/PlanCompiler/LimitExpressionTests.cs

```diff
@@ -834,6 +836,42 @@ orderby building.Address.ZipCode
         }
 
         [Fact]
+        public void Limit_ComplexModel_OrderBy_Skip_without_rownumber()
+        {
+            var log = new StringWriter();
+            using (var context = new AdvancedPatternsMasterContext())
+            {
+                var previousFlag = SqlProviderServices.UseOrderByRowNumberForQueriesWithOffset;
+                try
+                {
+                    SqlProviderServices.UseOrderByRowNumberForQueriesWithOffset = false;
```

Comment:
also, the flag is not part of the query cache key, so if one runs a query with flag set to true (default), then changes the flag to false and rerun the query, we will issue the same query - are we ok with this?